packages/contact-manager-rest-api/src/api-functions.ts

```diff
@@ -0,0 +1,127 @@
+import { checkAuth, sendErrorResponse, ISingleEndpointOpts } from '@sphereon/ssi-express-support'
+import { Request, Response, Router } from 'express'
+import { IRequiredContext } from './types'
+import Debug from 'debug'
+import { AddContactArgs } from '@sphereon/ssi-sdk.contact-manager'
+
+const debug = Debug('sphereon:ssi-sdk:contact-manager-rest-api')
+
+export function partyReadEndpoints(router: Router, context: IRequiredContext, opts?: ISingleEndpointOpts) {
+  if (opts?.enabled === false) {
+    debug(`Endpoint is disabled`)
+    return
+  }
+  const path = opts?.path ?? '/parties'
+  router.get(path, checkAuth(opts?.endpoint), async (request: Request, response: Response) => {
+    try {
+      // later we will add filter to this
+      const parties = await context.agent.cmGetContacts()
+      response.statusCode = 200
+      return response.send(parties)
+    } catch (e) {
+      return sendErrorResponse(response, 500, e.message as string, e)
+    }
+  })
+  router.get(`${path}/:partyId`, checkAuth(opts?.endpoint), async (request: Request, response: Response) => {
+    if (opts?.enabled === false) {
+      debug(`Endpoint is disabled`)
+      return
+    }
+    try {
+      const partyId = request.params.contactId
+      const party = await context.agent.cmGetContact({ contactId: partyId })
+      response.statusCode = 200
+      return response.send(party)
+    } catch (e) {
+      console.error(e)
+      return sendErrorResponse(response, 500, e.message as string, e)
+    }
+  })
+}
+
+export function partyWriteEndpoints(router: Router, context: IRequiredContext, opts?: ISingleEndpointOpts) {
+  if (opts?.enabled === false) {
+    debug(`Endpoint is disabled`)
+    return
+  }
+  const path = opts?.path ?? '/parties'
+  router.post(path, async (request: Request, response: Response) => {
+    try {
+      const addParty = request.body
+      const party = await context.agent.cmAddContact(addParty as AddContactArgs)
+      response.statusCode = 201
+      return response.send(party)
+    } catch (error) {
+      console.error(error)
```

Comment:
No console.errors. The sendErrorResponse already takes care of that

Reply:
removed